security/ssh_certificate.ksy

```diff
@@ -0,0 +1,196 @@
+        enum: cert_type
+      - id: key_id
+        type: cstring_utf8
+      - id: valid_principals
+        doc: | 
+          As a special case, a zero-length "valid principals"  field means
+          the certificate is valid for any principal of the specified type.
+        type: packed_cstring
+      - id: valid_after
+        doc: Unix timestamp (seconds since 1970-01-01 00:00:00)
+        type: u8
+      - id: valid_before
+        doc: | 
+          Unix timestamp (seconds since 1970-01-01 00:00:00)
+          When "forever" is requested this is set to 0xFFFF_FFFF_FFFF_FFFF
+        type: u8
+      - id: critical_options
+        doc: | 
+          critical options is a set of zero or more key options encoded as
+          below. All such options are "critical" in the sense that an implementation
+          must refuse to authorise a key that has an unrecognised option.
+
+          Generally, critical options are used to control features that restrict
+          access where extensions are used to enable features that grant access.
+          This ensures that certificates containing unknown restrictions do not
+          inadvertently grant access while allowing new protocol features to be
+          enabled via extensions without breaking certificates' backwards
+          compatibility.
+        type: packed_cstring_tuple
+      - id: extensions
+        doc: | 
+          extensions is a set of zero or more optional extensions. These extensions
+          are not critical, and an implementation that encounters one that it does
+          not recognise may safely ignore it.
+        type: packed_cstring_tuple
+      - id: reserved
+        doc: Unused currently
+        type: cstring_bytes
+      - id: signature_key
+        doc: | 
+          The signature key field contains the CA key used to sign the
+          certificate. The valid key types for CA keys are ssh-rsa,
+          ssh-dss, ssh-ed25519 and the ECDSA types ecdsa-sha2-nistp256,
+          ecdsa-sha2-nistp384, ecdsa-sha2-nistp521. "Chained" certificates, where
+          the signature key type is a certificate type itself are NOT supported.
+          Note that it is possible for a RSA certificate key to be signed by a
+          Ed25519 or ECDSA CA key and vice-versa.
+        type: cstring_sshkey
+      - id: signature
+        doc: | 
+          signature is computed over all preceding fields from the initial string
+          up to, and including the signature key. Signatures are computed and
+          encoded according to the rules defined for the CA's public key algorithm
+          (RFC4253 section 6.6 for ssh-rsa and ssh-dss, RFC5656 for the ECDSA
+          types), and draft-josefsson-eddsa-ed25519-03 for Ed25519.
+        type: packed_cstring_tuple
+    enums:
+      cert_type:
+        1: user
+        2: host
+  packed_cstring:
+    seq:
+      - id: len
+        type: u4
+      - id: packed_strings
+        type: cstrings
+        size: len
```

Comment:
Strictly speaking, `size` should be above `type` (https://doc.kaitai.io/ksy_style_guide.html#seq-attr):

```suggestion
      - id: packed_strings
        size: len
        type: cstrings
```

Also, consider renaming all `len` properties to `len_{FIELD}`, where `{FIELD}` is the identifier of a field that is delimited by the `len` size (e.g. `len` -> `len_packed_strings`, `len_value`, etc.). See https://doc.kaitai.io/ksy_style_guide.html#attr-id:

> * For fields that designate **size** of some particular data structure (in bytes or some other fixed units), use `len_` prefix and name of that data structure — i.e. `len_block` (length of a single block entry), `len_blocks` (total length of whole `blocks` array, made of `block` entries).